probe/proc/reader_linux.go

```diff
@@ -0,0 +1,203 @@
+package proc
+
+import (
+	"bytes"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/bluele/gcache"
+)
+
+const (
+	filesCacheLen        = 512
+	filesCacheExpiration = 60 * time.Second
```

Comment:
Yes, it is a cache for `/proc/something`, but for file handles, not for content. The most common case is when we read from `/proc/<some_proc_id>/[comm, cmdline, stat]`: we try to get from the cache and hopefully we will get an already-open file handle. A `60 * time.Second` expiration time would mean that we could have entries in the cache for up to 60 seconds that will never be used...

Further improvements would be to cache the content for `comm` and `cmdline`, and we could even cache the `stat` info for `/proc/<some_proc_id>/fd/*`.